lnd_test.go

```diff
@@ -9118,6 +9115,20 @@ func testMultiHopReceiverChainClaim(net *lntest.NetworkHarness, t *harnessTest)
 		t.Fatalf(predErr.Error())
 	}
 
+	// The invoice should show as settled for Carol, even if it was swept
+	// on-chain.
+	invoicesReq := &lnrpc.ListInvoiceRequest{}
+	invoicesResp, err := carol.ListInvoices(ctxb, invoicesReq)
+	if err != nil {
+		t.Fatalf("unable to retrieve invoices: %v", err)
+	}
+	if len(invoicesResp.Invoices) != 1 {
+		t.Fatalf("expected 1 invoice, got %d", len(invoicesResp.Invoices))
+	}
+	if !invoicesResp.Invoices[0].Settled {
```

Comment:
For future proofing, it should start to use the new invoice state field recently added.

Reply:
Fixed.